### ISSUE TITLE
3D Structures Tab

### ISSUE THREAD

**Issue by jessicagood** [StraboSpot/strabo-mobile#202]
This tab loads very slowly and I would like to organize it differently. What does anyone think of having this tab look and act similar to the new Sample, Other Features and Orientation tabs?

The format used for those tabs would allow the user to add more than one 3D Structure. But is this a valid case? @zmichels @JEAndrew @panamint55

**Comment by jessica-azgs** [comment]
Can our main list of 3D Structures be broken into 2-4 different types of 3D Structures and then with select lists narrowed down to its more specific type? The list we have now has 10 types: Folds, Boudinage, Tectonites, Mullions, Lobate-Cuspate, Igneous/Migmatite, Elliposoidal Data, Non-Ellipsoidal Data, Elliptical Data and Other 3-D Structure.

I'm trying to make this tab more like the Orientation tab but was hoping to avoid having 10 different buttons at the top of the tab for each of these types of 3D Structures.

**Comment by zmichels** [comment]
Just talked with Basil and Maureen quickly about this.
Basil doesn't really like having just "Add a 3D structure" button, and would prefer to see the options we currently list. I am less adverse to this idea, and would be fine with the approach you suggested.
He mentioned that perhaps instead of a toggle switch, maybe "Add a...__" button for each.

**Comment by jessica-azgs** [comment]
Right, I don't want a single button for all of 3D Structures either. Too much skip logic to deal with and that would make it load slower. The other alternative is do an Add a Fold button, Add a Boudinage button, etc. That just ends up being 10 different buttons - which is a lot and won't look that great.

What I would like to do is have 2-4 'Add a SOMETHING' buttons. When you click that button the first field has you narrow down the feature more specifically.

If this list can't be made into 2-4 types I'll do the 10. But once I do it one way it will take more work if we want to switch to another method.

**Comment by panamint55** [comment]
I am ok with the add a 3-d structure.  It makes the purpose of the tab a little less clear, folks will have to explore this, but is probably fine as the first time they use it.

Doug


***********************************************
J. Douglas Walker
Isotope Geochemistry Laboratories
Nichols Hall
2335 Irving Hill Drive
Lawrence, KS  66045
785-864-7711
785-864-7789 (fax)
***********************************************

**Comment by jessica-azgs** [comment]
There are too many fields and too much skip logic to have a single 'Add 3D Structure' button. It needs to be broken down some. I was just hoping that there might be a logical way it could be broken into 2-4 types instead of the 10 we have now.

**Comment by panamint55** [comment]
From:  Folds, Boudinage, Tectonites, Mullions, Lobate-Cuspate, Igneous/Migmatite, Elliposoidal Data, Non-Ellipsoidal Data, Elliptical Data and Other 3-D Structure.

To:  Folds, Boudinage, Tectonites, Mullions/Lobate-Cuspate, Igneous, Strain, and Other?

Just an idea

Doug

***********************************************
J. Douglas Walker
Isotope Geochemistry Laboratories
Nichols Hall
2335 Irving Hill Drive
Lawrence, KS  66045
785-864-7711
785-864-7789 (fax)
***********************************************

**Comment by zmichels** [comment]
Just passing on another idea from Basil:

**Fabric:**
    tectonite
    igneous
    migmatite
**Fold:**
**Boudins / Mullions / Lobate-cuspate:**
**Elliptical / Ellipsoidal data:**

**Comment by jessica-azgs** [comment]
Least that's down to 4 though some of the type names are really long. I think I'll let you guys discuss that tomorrow and implement it when I get back next Tuesday. After you all decide how you want it if someone could split the 3D Structures form into new KoBo forms, one each for the 2-4 main types (like I did for Orientations), that would be helpful. And then use select lists instead of acknowledge fields to narrow down to the more specific structure. Thanks!

**Comment by panamint55** [comment]
Sounds good.

Doug

***********************************************
J. Douglas Walker
Isotope Geochemistry Laboratories
Nichols Hall
2335 Irving Hill Drive
Lawrence, KS  66045
785-864-7711
785-864-7789 (fax)
***********************************************

**Comment by zmichels** [comment]
We talked a bit today about this.

The general agreement is that yes, we should group the options so there are fewer to start. So far, no real objections to the list I proposed in my last post, but we (in Wisconsin) are going to think more thoroughly about the list before next week.

Two main points that came up a couple times: [1] we would like to see this tab be as much like the orientations tab page as possible; and [2] we want to be able to add multiple of each (like we can on orientations page).

Z

**Comment by jessica-azgs** [comment]
@zmichels Any other thoughts on this so I can redo the page? And can you or @JEAndrew make them into 4 separate KoBo forms (for Fabric, Fold, Boudins / Mullions / Lobate-cuspate and Elliptical / Ellipsoidal data or whatever top level categories you come up with)? Boudins / Mullions / Lobate-cuspate is a really long name though so if there is a another word/phrase you could use that would encompass those that would look a lot better.

**Comment by JEAndrew** [comment]
OK, I broke up 3-D structures into 4 forms. Can someone check to make sure these are what we want?

**Comment by JEAndrew** [comment]
Jessica? I don't remember if you could use group skip logic or should I go through an make individual skip logic

**Comment by JEAndrew** [comment]
Updated Kobo forms. 3-D structures split into 4 forms.

**Comment by jessica-azgs** [comment]
1. You can tie skip logic to groups.
1. Did you mean to title the one form 'Ellip' or was it truncated accidentally?
1. Within each form I think you want the first field to be a select-box where the user selects if, for example, the specific 3D Structure they are adding is a Tectonite or a Mullion. The way you have it a user can have a single 3D Structure be both types. Where in fact I think we want them to add a 2nd 3D Structure if they want to to do both a Tectonite and a Mullion. (If that's even possible, but either way what I'm saying is applicable to all the top level fields of each new form.) So in this new way you wouldn't have the acknowledge fields at all for the top level fields, it'd just be a single select box.

**Comment by JEAndrew** [comment]
2. Yes, I meant Ellip. These letters are shared for the options in the form, otherwise the title would be long or difficult to explain. Anyone else have a better suggestion? I updated the name to Tensors, any better for the geologists?

**Comment by JEAndrew** [comment]
3. I have the forms set up this way you describe for everything except folds. Should I do this in the folds form so the user can enter a second fold?

**Comment by jessica-azgs** [comment]
I need to do the button to enter a 2nd fold. That shouldn't be in the KoBo Forms.

Otherwise the forms looks the same to me. They haven't been changed to how I outlined in #3 above, or maybe you are still working on them? Each form needs to start with a select box, not an acknowledge field.

**Comment by JEAndrew** [comment]
Select multiple box?

**Comment by jessica-azgs** [comment]
I did the start of Fabrics for you. Check that out. You start with a select box and select the type of Fabric you are entering. If you select Tectonite then the Skip Logic opens more fields for tectonites. You can only enter one type of fabric, so no it is not a select-mulitple. I will write to code to handle entering a 2nd type of fabic (or a 2nd type of any 3D structure) - that part won't be in KoBo.

**Comment by JEAndrew** [comment]
Ok, I updated these per your example

**Comment by jessica-azgs** [comment]
@JEAndrew Can we just call the `Boudin-Mullion-3D` form `Other 3D Structures`?

**Comment by jessica-azgs** [comment]
@JEAndrew I think you need some skip logic in there tied to the new top-level select boxes. Make sure any skip logic is tied to an actual field and doesn't just show because a group is open. For example, the skip logic for **each** field in Ellipsoidal Tensor Data needs to be tied to the top-level Tensor Type field.

**Comment by jessica-azgs** [comment]
I added a label field to the top of each form so the user can identify each 3D Structure with a personalized label. A default label is filled in in the app if none is given.

I also changed the `data column name` for the top level select box to `feature_type` to make it consistent with all of the orientation forms. Folds does not have a top level select box which makes a few more challenges since it's not consistent with the other 3D Structures forms. I have overcome that hurdle for now so I think it's ok. But if you did want to add a top level select box to Folds for different types of folds and call the `data column name` for that field `feature_type` that would work too and improve consistency.

**Comment by JEAndrew** [comment]
1. updated the name boudin.... to other 3d structure
2. Changed top level of folds for fold type

Not done yet: Updating the skip logic for the within group fields will me a while to do.

**Comment by jessica-azgs** [comment]
1. I added the Label field back to the top of Fold. Thought it was there before but didn't see it when I looked just now.
2. Also for Fold Type I changed the Data Column Name (seen when you go to the settings for that field) to `feature_type` just to be consistent with the other forms.
3. Yeah, unfortunately you're right, the skip logic will take some time. Let me know when you're done and I'll pull it in to the app.
